bolero-model/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+// WIP
+
+// use crate::{fuzz, generator::*};
+// use arrayvec::ArrayVec;
+use bolero_generator::ValueGenerator;
+// use std::collections::LinkedList;
+
+// #[derive(Debug, TypeGenerator)]
+// enum Operation {
+//     Push(u8),
+//     Pop,
+//     Clear,
+// }
+
+// #[derive(Default)]
+// struct Model {
+//     subject: ArrayVec<[u8; 32]>,
+//     oracle: LinkedList<u8>,
+// }
+
+// #[model]
+// impl Model {
+//     #[new]
+//     fn new() -> Self {
+//         Self::default()
+//     }
+
+//     #[reset]
+//     fn reset(&mut self) {
+//         self.clear()
+//     }
+
+//     fn push(&mut self, value: u8) {
+//         self.subject.push(value);
+//         self.oracle.push_front(value);
+//     }
+
+//     fn pop(&mut self) {
+//         let actual = self.subject.pop();
+//         let expected = self.oracle.pop_front();
+//         assert_eq!(actual, expected);
+//     }
+
+//     fn clear(&mut self) {
+//         self.subject.clear();
+//         self.oracle.clear();
+//     }
+// }
+
+pub trait FuzzModel {
+    type Arguments;
+    type ArgumentsGen: ValueGenerator<Output = Self::Arguments>;
+    type Operation;
+    type OperationGen: ValueGenerator<Output = Self::Operation>;
+
+    fn gen_arguments() -> Self::ArgumentsGen;
+    fn gen_operation() -> Self::OperationGen;
+
+    fn init(arguments: Self::Arguments);
+    fn dispatch(&mut self, operation: Self::Operation);
+}
+
+// fn main() {
+//     fuzz!().model::<Model>();
+// }
```

Comment:
Remove?

Reply:
That's just an idea of what the interface would be like. The whole bolero-model crate is WIP so I'll just skip pushing it for now.

Reply:
I like this interface a _lot_ though.